operator/builtin/input/azure/loganalytics/log_analytics.go

```diff
@@ -0,0 +1,113 @@
+	"fmt"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/builtin/input/azure"
+	"github.com/observiq/stanza/operator/helper"
+)
+
+const operatorName = "azure_log_analytics_input"
+
+func init() {
+	operator.Register(operatorName, func() operator.Builder { return NewLogAnalyticsConfig("") })
+}
+
+// NewLogAnalyticsConfig creates a new Azure Log Analytics input config with default values
+func NewLogAnalyticsConfig(operatorID string) *LogAnalyticsInputConfig {
+	return &LogAnalyticsInputConfig{
+		InputConfig:   helper.NewInputConfig(operatorID, operatorName),
+		PrefetchCount: 1000,
+		StartAt:       "end",
+	}
+}
+
+// LogAnalyticsInputConfig is the configuration of a Azure Log Analytics input operator.
+type LogAnalyticsInputConfig struct {
+	helper.InputConfig `yaml:",inline"`
+
+	// required
+	Namespace        string `json:"namespace,omitempty"         yaml:"namespace,omitempty"`
+	Name             string `json:"name,omitempty"              yaml:"name,omitempty"`
+	Group            string `json:"group,omitempty"             yaml:"group,omitempty"`
+	ConnectionString string `json:"connection_string,omitempty" yaml:"connection_string,omitempty"`
+
+	// optional
+	PrefetchCount uint32 `json:"prefetch_count,omitempty" yaml:"prefetch_count,omitempty"`
+	StartAt       string `json:"start_at,omitempty"       yaml:"start_at,omitempty"`
+}
+
+// Build will build a Azure Log Analytics input operator.
+func (c *LogAnalyticsInputConfig) Build(buildContext operator.BuildContext) ([]operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(buildContext)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.Namespace == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'namespace'", operatorName)
+	}
+
+	if c.Name == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'name'", operatorName)
+	}
+
+	if c.Group == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'group'", operatorName)
+	}
+
+	if c.ConnectionString == "" {
+		return nil, fmt.Errorf("missing required %s parameter 'connection_string'", operatorName)
+	}
+
+	if c.PrefetchCount < 1 {
+		return nil, fmt.Errorf("invalid value '%d' for %s parameter 'prefetch_count'", c.PrefetchCount, operatorName)
+	}
+
+	var startAtEnd bool
+	switch c.StartAt {
+	case "beginning":
+		startAtEnd = false
+	case "end":
+		startAtEnd = true
+	default:
+		return nil, fmt.Errorf("invalid value '%s' for %s parameter 'start_at'", c.StartAt, operatorName)
+	}
+
+	logAnalyticsInput := &LogAnalyticsInput{
+		EventHub: azure.EventHub{
+			Namespace:     c.Namespace,
+			Name:          c.Name,
+			Group:         c.Group,
+			ConnStr:       c.ConnectionString,
+			PrefetchCount: c.PrefetchCount,
+			StartAtEnd:    startAtEnd,
+			Persist: &azure.Persister{
+				DB: helper.NewScopedDBPersister(buildContext.Database, c.ID()),
+			},
+			InputOperator: inputOperator,
+		},
+		json: jsoniter.ConfigFastest,
+	}
```

Comment:
Continuing from the above comment, this could all be replaced by:
```go
	azSettings, err := c.AzureConfig.Build(buildContext)
	if err != nil {
		return nil, err
	}

	logAnalyticsInput := &LogAnalyticsInput{
		EventHub: azure.EventHub{
			AzureSettings: azSettings,
			Persist: &azure.Persister{
				DB: helper.NewScopedDBPersister(buildContext.Database, c.ID()),
			},
			InputOperator: inputOperator,
		},
		json: jsoniter.ConfigFastest,
	}
```